### PR TITLE
[WiP] Old Guard NPCs

### PR DESCRIPTION
**Pull request by faefux**
#### Summary
Content "Basic Old Guard NPCs"

#### Purpose of change

Adding basic NPCs as per #49114
Todo/Done since uploading:

- [  ]DISREGARDED - Fix Traders inventory
- [x] Accurate pilot equipment loadout
   - [x] Add pilot clothes as per: #49119
- [ ] DISREGARDED - Accurate  equipment loadout
- [x] Added proficiencies `gunsmithing_basic` and `prof_spotting` to the NC_SOLDIER class. 
- [x] Add Officer NPC as camp leader.
  - [ ] Add dress-uniform  as per: #49128 
- [x] Write a handful of short old guard related [backrounds](https://github.com/CleverRaven/Cataclysm-DDA/tree/master/data/json/npcs/Backgrounds)
- [ ] DISREGARDED - Add other NPCs if the PR, life and time permits.

#### Describe the solution

Currently: A trader, some regular soldiers and two pilots.
The pilots are using the regular soldiers dialogue for now. which is just saying hello and politely to go away.

#### Describe alternatives you've considered

Writing full characters but I thought it would be best to get the bare bones up before getting lost in writing at least 6-8 fleshed out characters.

#### Testing

I've tested them by talking to them after they're spawned by a mission completion with everything pieced together. They seemed to work fine.

#### Additional context

Part of the reason I uploaded this so early as a draft was in the help others can aid with the trader.  Have not done a trader before or really payed much attention to those in game so it'd be great if others could help adjust there stock according to what's reasonable.

### CODE REVIEW

File: data/json/npcs/old_guard/NPC_Old_Guard_FOB_officer.json [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      { "item": "pencil", "count": 2 },
+      { "item": "pen" },
+      { "item": "rag" },
+      { "item": "red_pen" }
+    ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_OLD_GUARD_FOB_OFFICER",
+    "dynamic_line": [ "I don't have any use for you right now.  You will be informed if I need you, soldier." ],
```
Comment (Maddremor) [inline data/json/npcs/old_guard/NPC_Old_Guard_FOB_officer.json:70]: Is the player supposed to always be a member of the Old Guard at the point when they can first interact with this NPC?

Comment (PatrikLundell) [reply data/json/npcs/old_guard/NPC_Old_Guard_FOB_officer.json:70]: I believe so. The base is established as a result of the PC performing a quest for the Old Guard (clearing out the location), so they shouldn't set up shop until the PC has done that. However, it's not impossible that there is (or will be) a way to fall out with the Old Guard so you no longer would be considered a member, but possibly not (yet) an enemy. I'd expect such an untrustworthy character wouldn't be "recruited" again, though.

Comment (faefux) [reply data/json/npcs/old_guard/NPC_Old_Guard_FOB_officer.json:70]: > the player supposed to always be a member of the Old Guard at the point when they can first interact with this NPC?
Nope. No real easy ways of doing this if its desirable even but I think that around the midpoint of OG tree you will start getting penalized for not doing some tasks in a reasonable timeframe to represent a greater trust placed in you and higher importance of of work given to the player.

**Comment by faefux** [comment]
Sorry that it is only tangentily related but I suspect it will help others too..  Can someone tel me what the eyes emoji means in your culture? 
It's nice being able to non-verbally show appreciation for work without or confirm things without having to write a sentence with the thumb up or heart but the rest I do not understand.  
Why are we trying to make CDDA github like reddit and discord?

**Comment by faefux** [comment]
I wrote some backrounds for old guard related soldiers and piots but am having trouble adding them, etc. can somebody explain to me or link me the docs on how the new backrounds system works a la https://github.com/CleverRaven/Cataclysm-DDA/tree/master/data/json/npcs/Backgrounds ? It doesn't seem so clear to me.

**Comment by PatrikLundell** [comment]
Well, faefux, you're not alone (but I'm incompetent at asocial media, so don't really count me).
Eyes: Don't know... Might be rolling, or might be surprise, or something else entirely (I don't think horror is likely, even if that's suitable from a technical standpont)...
Rocket: Don't know... Might be fireworks (in which case I don't really see any difference between that and applause), or something else...

**Comment by faefux** [comment]
I'd say this was done more or less. Was interested in doing the other things but they are outside my scope of understanding and nobody else seems interested in filling the gaps.

**Comment by actual-nh** [comment]
> Sorry that it is only tangentily related but I suspect it will help others too.. Can someone tel me what the eyes emoji means in your culture?

I've most often seen it used for "am taking a look" or "will take a look at".

> Why are we trying to make CDDA github like reddit and discord?

Point, although I actually find github's few emoji easier to deal with.

**Comment by faefux** [comment]
Sincerely I will return to this, It seems a shame to not finish giving the Old Guard a little present in the world.

**Comment by I-am-Erk** [comment]
Closing as abandoned, bookmarking in the hopes you or someone will resume this

**Comment by LyleSY** [comment]
Is anything missing here? This all looks fine to me

**Comment by I-am-Erk** [comment]
It was draft, which implies incomplete. I haven't looked through in detail to assess where there could be flaws. You are welcome to give it a test run and, if it's working, either open a new pr with these contents or reopen this one and work it out.

**Pull request closed without merging**